python/sglang/srt/checkpoint_engine/checkpoint_engine_worker.py

```diff
@@ -128,9 +142,14 @@ def post_hook():
                     quant_method = getattr(module, "quant_method", None)
                     if quant_method is not None:
                         # Move parameters to device if needed for quantization processing
-                        target_device = torch.device(
-                            "cuda", torch.cuda.current_device()
-                        )
+                        if is_npu:
+                            target_device = torch.device(
+                                "cuda", torch.cuda.current_device()
+                            )
+                        else:
+                            target_device = torch.device(
+                                "npu", torch_npu.npu.current_device()
+                            )
```

Comment:
I'm unsure whether the checkpoint engine representation needs modifications for NPU compatibility.